### ISSUE TITLE
Umbraco 8 Package install, picks the Default Package not the v8 one

### ISSUE THREAD

**Issue by KevinJump** [umbraco/Umbraco-CMS#4363]
Hi, 
_This might be a mixture of Umbraco 8 or our.umbraco.com Issue - so Just throwing this in, as it will be more of an issue as more packages are added._ 

You can't as a package developer mark two packages as current (one for v7, one for v8) - 
and in Umbraco 8 - when installing a package from the package section it picks the current one (not the v8.0 one) from the list of packages. 

**Either**
Our only lets you mark one package as current - so it might be that our has to let you mark active packages based on version. 

**or ?** 
Umbraco 8 doesn't look for the package from the list that is v8 so it might be that the package installer picks the correct package if there is one.

**Comment by Shazwazza** [comment]
The package system has a 'minimum' required umbraco version. This exists both in v7 and v8 and it's not changed in v8. And just like any major version upgrade, you would need to version your own software accordingly.

For example, I would have Articulate 3.x which is compatible with Umbraco 7.x and Articulate 4.x which is compatible with Umbraco 8.x. It is not possible in any software maintenance and version releases with regards to breaking changes to have the same software version compatible with 2 or more different incompatible target software versions.

When you create packages for Umbraco and you want to have it support both 7.x and 8.x then you would make two different major versions of your package, each targeting the min umbraco version.

The package system in the back office when it searches should only return the results of compatible packages and only install the latest compatible version. If this isn't working correctly then it's an issue with Our and will need to be investigated. Marking packages as active on OUR isn't actually used in this algorithm.

**Comment by nul800sebastiaan** [comment]
Yep, I think you're both saying the same thing. The problem seems to be that v8 tries to install the version marked as "Current" instead of the version that's compatible with v8.

![image](https://user-images.githubusercontent.com/304656/52208858-4f220580-2882-11e9-9686-5f82527bcc8d.png)

There's 2 versions available, v8 should figure out that the v7 version would not be compatible and can't be installed. I guess we need to have a "max" version as well as a minumum version.

**Comment by Shazwazza** [comment]
It's not v8 that does this logic, its Our. Umbraco just passes up it's version to Our and Our figures it out. We don't need a 'max' since the way it works is:

If i have 2x package files and they are flagged with min versions like 7.9 and 8.0, then these version numbers are sorted correctly, and the highest one is taken that is greater than the requested version, else nothing is returned. We could add more logic to say: ignore all major versions that don't match the current one too.

So something in that filtering logic on Our isn't doing the right thing and i suspect its because of the major version change - since we've never had to deal with that before.
